homeassistant/components/no_ip/config_flow.py

```diff
@@ -0,0 +1,225 @@
+"""Adds config flow for No-IP.com integration."""
+from __future__ import annotations
+
+import asyncio
+import base64
+import logging
+from typing import Any
+
+import aiohttp
+from aiohttp.hdrs import AUTHORIZATION, USER_AGENT
+import async_timeout
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import (
+    CONF_DOMAIN,
+    CONF_IP_ADDRESS,
+    CONF_PASSWORD,
+    CONF_SCAN_INTERVAL,
+    CONF_TIMEOUT,
+    CONF_USERNAME,
+)
+from homeassistant.core import callback
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers import aiohttp_client
+from homeassistant.helpers.selector import (
+    NumberSelector,
+    NumberSelectorConfig,
+    NumberSelectorMode,
+    TextSelector,
+    TextSelectorConfig,
+    TextSelectorType,
+)
+
+from .const import (
+    DEFAULT_SCAN_INTERVAL,
+    DEFAULT_TIMEOUT,
+    DOMAIN,
+    HA_USER_AGENT,
+    MANUFACTURER,
+    NO_IP_ERRORS,
+    UPDATE_URL,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_SCHEMA = {
+    vol.Required(CONF_DOMAIN): TextSelector(
+        TextSelectorConfig(type=TextSelectorType.TEXT)
+    ),
+    vol.Required(CONF_USERNAME): TextSelector(
+        TextSelectorConfig(type=TextSelectorType.TEXT)
+    ),
+    vol.Required(CONF_PASSWORD): TextSelector(
+        TextSelectorConfig(type=TextSelectorType.PASSWORD)
+    ),
+    vol.Optional(CONF_TIMEOUT, default=DEFAULT_TIMEOUT): NumberSelector(
+        NumberSelectorConfig(
+            min=10,
+            max=9999,
+            mode=NumberSelectorMode.BOX,
+            unit_of_measurement="seconds",
+        ),
+    ),
```

Comment:
Why would this be configurable? It should have a sane default out of the box, let's not expose a setting like this.